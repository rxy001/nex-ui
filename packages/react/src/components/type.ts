export type ClassNames = string | string[]

export type ComponentClasses<OwnerState, T extends string> = Partial<
  Record<T, ClassNames | ((ownerState: OwnerState) => ClassNames)>
>
