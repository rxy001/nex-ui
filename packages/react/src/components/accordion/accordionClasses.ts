import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const accordionClasses = generateUtilityClasses('nui-accordion', [
  // ---------root---------
  'root',

  // "underlined" | "outlined"
  'variant-underlined',
  'variant-outlined',
])

export const accordionItemClasses = generateUtilityClasses(
  'nui-accordion-item',
  [
    // ---------root---------
    'root',
    'disabled',

    // ---------heading---------
    'heading',

    // ---------trigger---------
    'trigger',

    // ---------content---------
    'content',

    // ---------indicator---------
    'indicator',

    'expanded',
  ],
)
