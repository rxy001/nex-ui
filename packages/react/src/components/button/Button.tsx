import { forwardRef } from 'react'
import type { Ref } from 'react'
import { WaterWave } from '../waterWave'
import type { ButtonProps } from './types'
import { useButton } from './useButton'

export const Button = forwardRef<
  HTMLButtonElement | HTMLAnchorElement,
  ButtonProps
>((props, ref) => {
  const { loading } = props

  const { getRootProps, startIcon, endIcon } = useButton(props)

  const { type, href, disabled, children, ...rootProps } = getRootProps()

  const childNode =
    href !== undefined ? (
      <a
        ref={ref as Ref<HTMLAnchorElement>}
        data-disabled={disabled || null}
        href={href}
        {...rootProps}
      >
        {startIcon}
        {children}
        {endIcon}
      </a>
    ) : (
      <button
        type={type}
        ref={ref as Ref<HTMLButtonElement>}
        disabled={disabled}
        {...rootProps}
      >
        {startIcon}
        {children}
        {endIcon}
      </button>
    )

  return <WaterWave disabled={loading || disabled}>{childNode}</WaterWave>
})
