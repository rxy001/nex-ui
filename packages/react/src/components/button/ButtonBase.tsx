'use client'

import { nex } from '@nex-ui/styled'
import { useEvent, useFocusRing } from '@nex-ui/hooks'
import { useMemo } from 'react'
import { isFunction, mergeProps } from '@nex-ui/utils'
import type { ComponentProps, ElementType, KeyboardEvent } from 'react'
import type { Interpolation } from '@nex-ui/system'
import type { Overwrite } from '../../types/utils'

type ButtonBaseProps<RootComponent extends ElementType = 'button'> = Overwrite<
  ComponentProps<RootComponent>,
  {
    as?: RootComponent
    sx?: Interpolation
    href?: string
  }
>

export const ButtonBase = <RootComponent extends ElementType = 'button'>(
  inProps: ButtonBaseProps<RootComponent>,
) => {
  const props = inProps as ButtonBaseProps<'button'>
  const {
    as,
    role,
    children,
    disabled,
    type = 'button',
    tabIndex = 0,
    ...remainingProps
  } = props

  const rootElement =
    as !== undefined
      ? as
      : typeof props.href === 'string' && props.href
        ? 'a'
        : 'button'

  const { focusVisible, focusProps } = useFocusRing()

  const handleKeyDown = useEvent((event: KeyboardEvent<HTMLButtonElement>) => {
    // Limit the repeated triggering of the click event when the Enter key is pressed.
    if (
      focusVisible &&
      event.code === 'Enter' &&
      event.target === event.currentTarget &&
      (event.currentTarget.tagName === 'BUTTON' ||
        event.currentTarget.tagName === 'A')
    ) {
      event.preventDefault()
    }
  })

  const handleKeyUp = useEvent((event: KeyboardEvent<HTMLButtonElement>) => {
    // Keyboard accessibility for non interactive elements
    if (
      focusVisible &&
      event.target === event.currentTarget &&
      (event.code === 'Space' || event.code === 'Enter')
    ) {
      event.currentTarget.click()
    }
  })

  const ariaProps = useMemo(() => {
    if (rootElement !== 'button' && !isFunction(rootElement)) {
      return {
        tabIndex: disabled ? -1 : tabIndex,
        role: role ?? 'button',
        'aria-disabled': props['aria-disabled'] ?? (disabled || undefined),
      }
    }
    return {
      tabIndex: disabled ? -1 : tabIndex,
      type,
      disabled,
      role,
    }
  }, [disabled, props, role, rootElement, tabIndex, type])

  const mergedProps = useMemo(() => {
    return mergeProps(remainingProps, focusProps, ariaProps)
  }, [ariaProps, focusProps, remainingProps])

  return (
    <nex.button
      as={rootElement}
      onKeyUp={handleKeyUp}
      onKeyDown={handleKeyDown}
      {...mergedProps}
    >
      {children}
    </nex.button>
  )
}

ButtonBase.displayName = 'ButtonBase'
