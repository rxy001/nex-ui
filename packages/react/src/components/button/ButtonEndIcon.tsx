import { nex } from '@nex-ui/styled'
import { buttonEndIcon } from '@theme'
import { useNexContext } from '../provider'
import type { ButtonIconProps } from './types'

export const ButtonEndIcon = ({ children, size }: ButtonIconProps) => {
  const { prefix, styles } = useNexContext()
  return (
    <nex.span
      css={styles(buttonEndIcon)({
        size: size === 'sm' ? size : undefined,
      })}
      className={`${prefix}-end-icon`}
    >
      {children}
    </nex.span>
  )
}
