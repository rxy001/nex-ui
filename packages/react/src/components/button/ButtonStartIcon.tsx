import { nex } from '@nex-ui/styled'
import { buttonStartIcon } from '../../theme'
import { useNexContext } from '../provider'
import type { ButtonIconProps } from './types'

export const ButtonStartIcon = ({ children, size, spin }: ButtonIconProps) => {
  const { prefix, styles } = useNexContext()
  return (
    <nex.span
      css={styles(buttonStartIcon)({
        size: size === 'sm' ? size : undefined,
        spin,
      })}
      className={`${prefix}-start-icon`}
    >
      {children}
    </nex.span>
  )
}
