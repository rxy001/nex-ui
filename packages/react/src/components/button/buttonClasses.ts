import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const buttonClasses = generateUtilityClasses('nui-btn', [
  // ---------root---------
  'root',
  // "link" | "text" | "solid" | "outline"
  'variant-link',
  'variant-text',
  'variant-solid',
  'variant-outline',

  // "sm" | "md" | "lg" | "full"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'icon-only',
  'loading',
  'disabled',
  'block',
  // ---------root---------

  'icon',
  'start-icon',
  'end-icon',

  'icon-size-sm',
  'icon-size-md',
  'icon-size-lg',

  'icon-loading',
])
