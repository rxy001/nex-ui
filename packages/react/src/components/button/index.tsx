export { Button } from './Button'

export type { ButtonProps, ButtonPropsVariantOverrides } from './types'
