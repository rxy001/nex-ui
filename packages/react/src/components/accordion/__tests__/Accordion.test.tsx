import { createRef, useState } from 'react'
import { fireEvent } from '@testing-library/react'
import {
  renderWithNexUIProvider,
  mountTest,
  rootClassNameTest,
} from '~/tests/shared'
import { Accordion, AccordionItem } from '../index'
import { Button } from '../../button'
import { accordionClasses, accordionItemClasses } from '../accordionClasses'
import type { Key } from 'react'

const children = (
  <AccordionItem itemKey='1' title='Accordion 1' data-testid='accordion-item'>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
    malesuada lacus ex, sit amet blandit leo lobortis eget.
  </AccordionItem>
)

describe('Accordion', () => {
  mountTest(<Accordion>{children}</Accordion>)

  rootClassNameTest(Accordion, 'test-class')

  it('renders correctly', () => {
    const { container } = renderWithNexUIProvider(
      <Accordion>{children}</Accordion>,
    )
    expect(container.firstElementChild).toMatchSnapshot()
  })

  it('should forward ref to Accordion', () => {
    const ref = createRef<HTMLDivElement>()
    const { container } = renderWithNexUIProvider(
      <Accordion ref={ref} data-testid='accordion'>
        {children}
      </Accordion>,
    )
    const accordionElement = container.firstElementChild
    expect(ref.current).toBe(accordionElement)
  })

  it('should forward ref to AccordionItem', () => {
    const ref = createRef<HTMLDivElement>()
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion>
        <AccordionItem
          ref={ref}
          itemKey='1'
          title='Accordion 1'
          data-testid='accordion-item'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </AccordionItem>
      </Accordion>,
    )
    const accordionItemElement = getByTestId('accordion-item')
    expect(ref.current).toBe(accordionItemElement)
  })

  it('should forward slotProps to AccordionItem', () => {
    const slotProps = {
      heading: { className: 'test-heading' },
      indicator: { className: 'test-indicator' },
      trigger: { className: 'test-trigger' },
      content: { className: 'test-content' },
    }
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion>
        <AccordionItem
          itemKey='1'
          title='Accordion 1'
          slotProps={slotProps}
          data-testid='accordion-item'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </AccordionItem>
      </Accordion>,
    )

    const root = getByTestId('accordion-item')

    expect(root.querySelector(`.${accordionItemClasses.heading}`)).toHaveClass(
      slotProps.heading.className,
    )
    expect(
      root.querySelector(`.${accordionItemClasses.indicator}`),
    ).toHaveClass(slotProps.indicator.className)
    expect(root.querySelector(`.${accordionItemClasses.trigger}`)).toHaveClass(
      slotProps.trigger.className,
    )
    expect(root.querySelector(`.${accordionItemClasses.content}`)).toHaveClass(
      slotProps.content.className,
    )
  })

  it('should forward classes to AccordionItem', () => {
    const classes = {
      heading: 'test-heading',
      indicator: 'test-indicator',
      trigger: 'test-trigger',
      content: 'test-content',
    }
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion>
        <AccordionItem
          itemKey='1'
          title='Accordion 1'
          classes={classes}
          data-testid='accordion-item'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </AccordionItem>
      </Accordion>,
    )
    const root = getByTestId('accordion-item')
    expect(root.querySelector(`.${classes.heading}`)).toBeInTheDocument()
    expect(root.querySelector(`.${classes.indicator}`)).toBeInTheDocument()
    expect(root.querySelector(`.${classes.trigger}`)).toBeInTheDocument()
    expect(root.querySelector(`.${classes.content}`)).toBeInTheDocument()
  })

  it('should have correct class names', () => {
    const { container, getByTestId, rerender } = renderWithNexUIProvider(
      <Accordion>{children}</Accordion>,
    )
    const accordionRoot = container.firstElementChild
    expect(accordionRoot).toHaveClass(accordionClasses.root)
    expect(accordionRoot).toHaveClass(accordionClasses['variant-underlined'])

    rerender(<Accordion variant='outlined'>{children}</Accordion>)
    expect(accordionRoot).toHaveClass(accordionClasses['variant-outlined'])

    rerender(<Accordion expandedKeys={['1']}>{children}</Accordion>)
    const accordionItem = getByTestId('accordion-item')
    expect(accordionItem).toHaveClass(accordionItemClasses.root)
    expect(accordionItem).not.toHaveClass(accordionItemClasses.disabled)
    expect(accordionItem).toHaveClass(accordionItemClasses.expanded)
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.trigger}`),
    ).toBeInTheDocument()
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.heading}`),
    ).toBeInTheDocument()
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.content}`),
    ).toBeInTheDocument()
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.indicator}`),
    ).toBeInTheDocument()
  })

  it('should add the appropriate variant class to root element based on variant prop', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <>
        <Accordion variant='outlined' data-testid='variant-outlined'>
          Button
        </Accordion>
        <Accordion variant='underlined' data-testid='variant-underlined'>
          Button
        </Accordion>
      </>,
    )

    const outlinedAccordion = getByTestId('variant-outlined')
    const underlinedAccordion = getByTestId('variant-underlined')

    expect(outlinedAccordion).toHaveClass(accordionClasses['variant-outlined'])
    expect(underlinedAccordion).toHaveClass(
      accordionClasses['variant-underlined'],
    )
  })

  it('should handle defaultExpandedKeys prop', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion defaultExpandedKeys={['1']}>{children}</Accordion>,
    )
    const accordionItem = getByTestId('accordion-item')
    expect(accordionItem).toHaveClass(accordionItemClasses.expanded)
  })

  it('should be controlled by expandedKeys prop', () => {
    function ControlledAccordion() {
      const [expandedKeys, setExpandedKeys] = useState<Key[]>([])

      return (
        <>
          <Accordion
            expandedKeys={expandedKeys}
            onExpandedKeysChange={setExpandedKeys}
          >
            <AccordionItem
              itemKey='1'
              title='Accordion 1'
              data-testid='accordion-item'
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </AccordionItem>
          </Accordion>
          <Button data-testid='button' onClick={() => setExpandedKeys(['1'])}>
            Open Accordion 1
          </Button>
        </>
      )
    }

    const { getByTestId } = renderWithNexUIProvider(<ControlledAccordion />)
    const accordionItem = getByTestId('accordion-item')
    expect(accordionItem).not.toHaveClass(accordionItemClasses.expanded)

    const button = getByTestId('button')
    fireEvent.click(button)
    expect(accordionItem).toHaveClass(accordionItemClasses.expanded)
  })

  it('should always render the AccordionItem content when keepMounted is true', () => {
    const { getByTestId, rerender } = renderWithNexUIProvider(
      <Accordion keepMounted expandedKeys={['1']}>
        {children}
      </Accordion>,
    )
    const accordionItem = getByTestId('accordion-item')
    expect(accordionItem).toBeInTheDocument()

    rerender(
      <Accordion keepMounted expandedKeys={[]}>
        {children}
      </Accordion>,
    )
    expect(accordionItem).toBeInTheDocument()
  })

  it('should unmount AccordionItem content when collapsed and keepMounted is false', () => {
    const { getByTestId, rerender } = renderWithNexUIProvider(
      <Accordion keepMounted={false} expandedKeys={['1']}>
        {children}
      </Accordion>,
    )
    const accordionItem = getByTestId('accordion-item')
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.content}`),
    ).toBeInTheDocument()

    rerender(
      <Accordion keepMounted={false} expandedKeys={[]}>
        {children}
      </Accordion>,
    )
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.content}`),
    ).not.toBeInTheDocument()
  })

  it('should handle multiple prop correctly', () => {
    const arrayChildren = [
      <AccordionItem key='1' itemKey='1' title='Accordion 1'>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
        malesuada lacus ex, sit amet blandit leo lobortis eget.
      </AccordionItem>,
      <AccordionItem key='2' itemKey='2' title='Accordion 2'>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
        malesuada lacus ex, sit amet blandit leo lobortis eget.
      </AccordionItem>,
      <AccordionItem key='3' itemKey='3' title='Accordion 3'>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
        malesuada lacus ex, sit amet blandit leo lobortis eget.
      </AccordionItem>,
    ]
    const { rerender, container } = renderWithNexUIProvider(
      <Accordion multiple expandedKeys={['1']}>
        {arrayChildren}
      </Accordion>,
    )
    const root = container.firstElementChild

    expect(
      root?.querySelectorAll(`.${accordionItemClasses.expanded}`),
    ).toHaveLength(1)

    rerender(
      <Accordion multiple expandedKeys={['1', '2']}>
        {arrayChildren}
      </Accordion>,
    )
    expect(
      root?.querySelectorAll(`.${accordionItemClasses.expanded}`),
    ).toHaveLength(2)
  })

  it('should handle disabled prop correctly', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion disabled={true}>
        <AccordionItem
          itemKey='1'
          title='Accordion 1'
          data-testid='accordion-item-1'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </AccordionItem>
        <AccordionItem
          itemKey='2'
          title='Accordion 2'
          data-testid='accordion-item-2'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </AccordionItem>
      </Accordion>,
    )
    const accordionItem1 = getByTestId('accordion-item-1')
    expect(accordionItem1).toHaveClass(accordionItemClasses.disabled)

    const accordionItem2 = getByTestId('accordion-item-2')
    expect(accordionItem2).toHaveClass(accordionItemClasses.disabled)
  })

  it('should hide indicator when hideIndicator is true', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion hideIndicator={true}>{children}</Accordion>,
    )
    const accordionItem = getByTestId('accordion-item')
    expect(
      accordionItem.querySelector(`.${accordionItemClasses.indicator}`),
    ).not.toBeInTheDocument()
  })

  it('should customize indicator', () => {
    const customIndicator = <span data-testid='custom-indicator'>▼</span>
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion indicator={customIndicator}>{children}</Accordion>,
    )

    const indicator = getByTestId('custom-indicator')
    expect(indicator).toBeInTheDocument()
  })

  it('should call onExpandedKeysChange when expanded keys change', () => {
    const onExpandedKeysChange = jest.fn()
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion onExpandedKeysChange={onExpandedKeysChange}>
        {children}
      </Accordion>,
    )
    expect(onExpandedKeysChange).not.toHaveBeenCalled()

    const accordionItem = getByTestId('accordion-item')
    const trigger = accordionItem.querySelector(
      `.${accordionItemClasses.trigger}`,
    )
    fireEvent.click(trigger!)
    expect(onExpandedKeysChange).toHaveBeenCalled()
  })

  it('should warn when multiple is false and expandedKeys has more than one key', () => {
    const consoleWarnSpy = jest
      .spyOn(console, 'warn')
      .mockImplementation(() => {})

    renderWithNexUIProvider(
      <Accordion multiple={false} expandedKeys={['1', '2']}>
        {children}
      </Accordion>,
    )
    expect(consoleWarnSpy).toHaveBeenCalledWith(
      '[Nex UI] Accordion: The multiple prop of the Accordion is set to false, but the number of currently expanded items exceeds 1. Please check and set the appropriate props.',
    )

    consoleWarnSpy.mockRestore()
  })

  it('should toggle between expanded and collapsed states when clicking the trigger', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <Accordion defaultExpandedKeys={['1']} multiple>
        <AccordionItem
          key='1'
          itemKey='1'
          title='Accordion 1'
          data-testid='accordion-item-1'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
          malesuada lacus ex, sit amet blandit leo lobortis eget.
        </AccordionItem>
        <AccordionItem
          key='2'
          itemKey='2'
          title='Accordion 2'
          data-testid='accordion-item-2'
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
          malesuada lacus ex, sit amet blandit leo lobortis eget.
        </AccordionItem>
      </Accordion>,
    )
    const accordionItem1 = getByTestId('accordion-item-1')
    expect(accordionItem1).toHaveClass(accordionItemClasses.expanded)
    const trigger1 = accordionItem1.querySelector(
      `.${accordionItemClasses.trigger}`,
    )
    fireEvent.click(trigger1!)
    expect(accordionItem1).not.toHaveClass(accordionItemClasses.expanded)

    const accordionItem2 = getByTestId('accordion-item-2')
    expect(accordionItem2).not.toHaveClass(accordionItemClasses.expanded)
    const trigger2 = accordionItem2.querySelector(
      `.${accordionItemClasses.trigger}`,
    )
    fireEvent.click(trigger2!)
    expect(accordionItem2).toHaveClass(accordionItemClasses.expanded)

    fireEvent.click(trigger2!)
    expect(accordionItem2).not.toHaveClass(accordionItemClasses.expanded)
  })
})
