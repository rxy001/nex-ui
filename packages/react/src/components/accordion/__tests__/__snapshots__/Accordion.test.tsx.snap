// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render with default props 1`] = `
.emotion-0 {
  width: var(--nui-sizes-full);
}

.emotion-1 {
  border-bottom: 1px solid var(--nui-colors-gray-highlight);
}

.emotion-2 {
  margin: 0;
  min-height: var(--nui-sizes-12);
}

.emotion-3 {
  padding: 0;
  margin: 0;
  outline: none;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  position: relative;
  box-sizing: border-box;
  width: var(--nui-sizes-full);
  min-height: inherit;
  -webkit-padding-start: var(--nui-spaces-4);
  padding-inline-start: var(--nui-spaces-4);
  -webkit-padding-end: var(--nui-spaces-4);
  padding-inline-end: var(--nui-spaces-4);
  font-size: var(--nui-font-sizes-xl);
  font-weight: var(--nui-font-weights-normal);
}

.emotion-3[data-focus-visible=true]:not(:disabled, [data-disabled=true]) {
  outline: var(--nui-borders-focus-visible-outline);
}

.emotion-4 {
  font-size: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="nui-accordion-root nui-accordion-variant-underlined emotion-0"
>
  <div
    class="nui-accordion-item-root emotion-1"
    data-testid="accordion-item"
  >
    <h3
      class="nui-accordion-item-heading emotion-2"
    >
      <button
        aria-controls="panel-1-content"
        aria-expanded="false"
        class="nui-accordion-item-trigger emotion-3"
        id="«r5»"
        tabindex="0"
        type="button"
      >
        <span>
          Accordion 1
        </span>
        <span
          aria-hidden="true"
          class="nui-accordion-item-indicator emotion-4"
          style="transform: none;"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.43 8.512a.75.75 0 0 1 1.058-.081L12 14.012l6.512-5.581a.75.75 0 0 1 .976 1.138l-7 6a.75.75 0 0 1-.976 0l-7-6a.75.75 0 0 1-.081-1.057"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </h3>
  </div>
</div>
`;
